arch/generic/crc32_c.c

```diff
@@ -25,7 +18,7 @@ uint32_t PREFIX(crc32_c)(uint32_t crc, const uint8_t *buf, size_t len) {
         aligned_len = len - algn_diff;
         if(aligned_len > CHORBA_LARGE_THRESHOLD)
             c = crc32_chorba_118960_nondestructive(c, (z_word_t*) aligned_buf, aligned_len);
-#  if W == 8
+#  if OPTIMAL_CMP == 64
```

Comment:
It might be a good idea to split this into two defines - one to reflect the size of the registers, and one to reflect the availability of unaligned loads and stores.